common/src/main/java/com/wynntils/core/persisted/config/Config.java

```diff
@@ -28,90 +31,138 @@ public void touched() {
 
     @Override
     public void store(T value) {
-        this.value = value;
+        Managers.Persisted.setRaw(this, value);
         // For now, do not call touch() on configs
     }
 
-    <P extends Configurable & Translatable> void createConfigHolder(P parent, Field configField, Persisted configInfo) {
-        Type valueType = Managers.Json.getJsonValueType(configField);
-        String fieldName = configField.getName();
+    public T getValue() {
+        // FIXME: This is just an alias now, should be changed to get() everywhere
+        return get();
+    }
 
-        boolean visible = !(this instanceof HiddenConfig<?>);
+    // FIXME: Old ways of setting the value. These should be unified, but since
+    // they have slightly different semantics, let's do it carfeully step by step.
 
-        String i18nKey = configInfo.i18nKey();
+    public void setValue(T value) {
+        if (value == null && !getMetadata().isAllowNull()) {
+            WynntilsMod.warn("Trying to set null to config " + getJsonName() + ". Will be replaced by default.");
```

Comment:
This should also be fixed in the new manager, so we don't silently modify configs in an enum breaks..

Reply:
I'm not sure I fully understand this. Do you mean that this piece of code is not very good, and we should fix it going forwards, so we do not reset configs to default? Or are you requesting a change in this PR?

Reply:
I would like this to be fixed in the PR that integrates this whole system. But since we are doing it incrementally, I am not sure when that is. Either way, I think this just corrupts configs. Either fix it here or later, I don't mind either way.